src/app/DeferredAttributePersistenceProvider.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AttributePersistenceProvider.h>
+#include <lib/support/ScopedBuffer.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace app {
+
+class DeferredAttribute
+{
+public:
+    explicit DeferredAttribute(const ConcreteAttributePath & path) : mPath(path) {}
```

Comment:
I guess this needs to be a concrete path because if we supported some sort of wildcarding here we'd run into issues with how to store the values?

And this can't be constexpr because the ConcreteAttributePath constructor is not constexpr, I guess?  And even if it were, since we store state we couldn't make the `DeferredAttribute` instances const anyway?

Reply:
Yes, initially I had a more complex implementation which allocated write buffer entries dynamically and put them in intrusive list, but later I thought that in typical use cases there should be few attributes with deferred writes, so I thought the flexibility would not justify the increased flash cost. But I guess I'm OK to change it if it comes handy.

`constexpr` doesn't seem to be useful for this class, yeah.